api/dataset_api.go

```diff
@@ -53,94 +54,108 @@ type JobLinks struct {
 }
 
 // GetInstance asks the Dataset API for the details for instanceID
-func (api *DatasetAPI) GetInstance(instanceID string) (Instance, error) {
+func (api *DatasetAPI) GetInstance(ctx context.Context, instanceID string) (instance Instance, err error, isFatal bool) {
 	path := api.url + "/instances/" + instanceID
 	logData := log.Data{"path": path, "instanceID": instanceID}
-	jsonBody, httpCode, err := api.get(path, nil)
+	jsonBody, httpCode, err := api.get(ctx, path, nil)
 	logData["httpCode"] = httpCode
-	if httpCode == http.StatusNotFound {
-		return Instance{}, nil
-	}
 	if err == nil && httpCode != http.StatusOK {
 		err = errors.New("Bad response while getting dataset instance")
+		if httpCode < http.StatusInternalServerError {
+			isFatal = true
+		}
+	} else if err != nil {
```

Comment:
same here

Reply:
it does get logged, a bit later - I've added a comment to that effect